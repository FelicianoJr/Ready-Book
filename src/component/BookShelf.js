import React from 'react'
import BookDetail from './BookDetail'

const BookShelf = ({ books = [], update, title }) => {

    const currentBooks = books.filter(_ => _.shelf === title);

    const separateTitle = title.replace(/([A-Z])/g, ' $1').trim();
    
    const titleFinal = separateTitle.charAt(0).toUpperCase() + separateTitle.slice(1);

    return (<div className="bookshelf">
        <h2 className="bookshelf-title">{titleFinal}</h2>
        <div className="bookshelf-books">
            <ol className="books-grid">
                {currentBooks.map(book => {
                    return <BookDetail key={book.id} book={book} update={update} />
                })}
            </ol>
        </div>
    </div>
    )
};

export default BookShelf;