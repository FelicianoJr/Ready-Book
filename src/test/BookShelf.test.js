import BookShelf from '../component/BookShelf'
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("<BookShelf/>", () => {
    
    let update;
    let props;

    beforeEach(() => {
        update = jest.fn();
        props = {
            books: [{
                id: 0, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React', shelf: 'read'
            },
            {
                id: 1, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React Native', shelf: 'read'
            }],
            title: 'Read',
            update: update,
        };
    })

    it("should contain the title passed as props", () => {
        const wrapper = mount(<BookShelf {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.props().title).toEqual('Read');
    });

    it("should contain length equal two in props array", () => {
        const wrapper = mount(<BookShelf {...props} />)
        expect(wrapper.props().books.length).toEqual(2);
    });

    it('should render two <div>', () => {
        const wrapper = shallow(<BookShelf {...props} />)
        expect(wrapper.find('div').length).toEqual(2);
    });

    it('should render one <ol>', () => {
        const wrapper = shallow(<BookShelf {...props} />)
        expect(wrapper.find('ol').length).toEqual(1);
    });

})