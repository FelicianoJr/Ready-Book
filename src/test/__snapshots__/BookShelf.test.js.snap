exports[`<BookShelf/> should contain the title passed as props 1`] = `
<BookShelf
  books={
    Array [
      Object {
        "id": 0,
        "imageLinks": Object {
          "thumbnail": "url/test/enzime",
        },
        "shelf": "read",
        "title": "React",
      },
      Object {
        "id": 1,
        "imageLinks": Object {
          "thumbnail": "url/test/enzime",
        },
        "shelf": "read",
        "title": "React Native",
      },
    ]
  }
  title="Read"
  update={[Function]}>
  <div
    className="bookshelf">
    <h2
      className="bookshelf-title">
      Read
    </h2>
    <div
      className="bookshelf-books">
      <ol
        className="books-grid" />
    </div>
  </div>
</BookShelf>
`;
